tokio/src/sync/mpsc/unbounded.rs

```diff
@@ -588,13 +598,29 @@ impl<T> Clone for WeakUnboundedSender<T> {
     }
 }
 
+impl<T> Drop for WeakUnboundedSender<T> {
+    fn drop(&mut self) {
+        self.chan.decrement_weak_count();
+    }
+}
```

Comment:
Do you not also need an increment in clone?

Please add a test that would have caught this.

Reply:
You're right.

I modified the test and can confirm that it fails before the fix.